src/btree/bt_cursor.c

```diff
@@ -9,6 +9,66 @@
 #include "wt_internal.h"
 
 /*
+ * WT_PAGE_PINNED --
+ *     If we have a page pinned and it's not been flagged for forced eviction
+ * (the latter test is so we periodically release pages grown too large).
+ */
+#define	WT_PAGE_PINNED(cbt)						\
+	(F_ISSET((cbt), WT_CBT_ACTIVE) &&				\
+	    (cbt)->ref->page->read_gen != WT_READGEN_OLDEST)
+
+/*
+ * WT_CURFILE_OP_XXX
+ *	If we're going to return an error, we need to restore the cursor to
+ * a valid state, the upper-level cursor code is likely to retry. The macros
+ * here are called to save and restore that state.
+ */
+#define	WT_CURFILE_OP_DECL						\
```

Comment:
I will inline `WT_PAGE_PINNED`.

The `WT_CURFILE_OP_XXX` #defines can't easily be inline functions because they declare variables scoped for the entire calling function, similar to the macros in `api.h`.